core/trino-main/src/main/java/io/trino/metadata/TableMetadata.java

```diff
@@ -27,7 +29,8 @@
     private final CatalogName catalogName;
     private final ConnectorTableMetadata metadata;
 
-    public TableMetadata(CatalogName catalogName, ConnectorTableMetadata metadata)
+    @JsonCreator
+    public TableMetadata(@JsonProperty("catalogName") CatalogName catalogName, @JsonProperty("connectorMetadata") ConnectorTableMetadata metadata)
```

Comment:
Does this object have to be Json serializable? The `ConnectorTableMetadata` is not json serializable.

Reply:
Yes, it should be serializable. It is not completely serializable. But, for now the the object can be deserialized:
```
    @JsonCreator
    public ConnectorTableMetadata(@JsonProperty("schemaTableName") SchemaTableName table, @JsonProperty("columns") List<ColumnMetadata> columns)
```
Other fields will be `null`. It works now because we do not use this fields. However, I agree that it is better to have truly serializable object. But, how to safely serialize `Map<String, Object>`? I am not sure whether it is possible. And this is why some fields were "skipped". Other fields could be easily serialized. 
Probably, it is possible to write a dedicated Serizalizer that is able to serialize `properties` (knowing which objects could be put into a `properties` map. But, should we really do it?

Reply:
>  But, how to safely serialize Map<String, Object>? I am not sure whether it is possible.

see `io.trino.plugin.jdbc.QueryParameter` serialization.

still, sounds like a problem we may not need to solve

Reply:
@findepi , thanks. So wait for other reviewers.